ranger/data/scope.sh

```diff
@@ -75,20 +75,53 @@ handle_extension() {
         odt|ods|odp|sxw)
             # Preview as text conversion
             odt2txt "${FILE_PATH}" && exit 5
+	    # Preview as markdown conversion
+	    pandoc -s -t markdown -- "${FILE_PATH}" && exit 5
             exit 1;;
 
+	# RTF and DOC
+	rtf|doc)
+	    # Preview as text conversion
+	    # Note: catdoc does not always work for .doc files
+	    catdoc -- "${FILE_PATH}" && exit 5
+	    exit 1;;
+
+	# DOCX, ePub, FB2 (using markdown)
+	# You might want to remove "|epub" and/or "|fb2" below if you have uncommented other methods to preview those formats
+	docx|epub|fb2)
+	    # Preview as markdown conversion
+	    pandoc -s -t markdown -- "${FILE_PATH}" && exit 5
+	    exit 1;;
+
+	# XLSX
+	xlsx)
+	    # Preview as csv conversion
+	    # Uses: https://github.com/dilshod/xlsx2csv
+	    xlsx2csv -- "${FILE_PATH}" && exit 5
+	    exit 1;;
+
+	# XLS
+	xls)
+	    # Preview as csv conversion
+	    # xls2csv comes with catdoc: http://www.wagner.pp.ru/~vitus/software/catdoc/
+	    xls2csv -- "${FILE_PATH}" && exit 5
+	    exit 1;;
+
         # HTML
         htm|html|xhtml)
             # Preview as text conversion
             w3m -dump "${FILE_PATH}" && exit 5
             lynx -dump -- "${FILE_PATH}" && exit 5
             elinks -dump "${FILE_PATH}" && exit 5
-            ;; # Continue with next handler on failure
+	    pandoc -s -t markdown -- "${FILE_PATH}" && exit 5
+	    ;; # Continue with next handler on failure
+
         # JSON
         json)
             jq --color-output . "${FILE_PATH}" && exit 5
             python -m json.tool -- "${FILE_PATH}" && exit 5
-            ;;
+	    pandoc -s -t markdown -- "${FILE_PATH}" && exit 5
```

Comment:
I don't think json's an input format to pandoc, only an output format. Unless you tested it and it works?

Reply:
I have not tested it, but the manual says it is:

https://pandoc.org/MANUAL.html#general-options

Reply:
Must've read over that because that's what I checked. *Still* though, it only works for json that represents a pandoc AST. That's not a useful general json preview.

Reply:
Will take it out then.